gapir/cc/replay_connection.cpp

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (C) 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "replay_connection.h"
+
+#include <grpc++/grpc++.h>
+#include <memory>
+
+#include "gapir/replay_service/service.grpc.pb.h"
+
+namespace gapir {
+
+// ResourceRequest member methods
+
+ReplayConnection::ResourceRequest::~ResourceRequest() {
+  if (mProtoResourceRequest != nullptr) {
+    delete mProtoResourceRequest;
+  }
+}
+
+bool ReplayConnection::ResourceRequest::append(const std::string& id,
+                                               size_t size) {
+  if (mProtoResourceRequest == nullptr) {
+    return false;
+  }
+  mProtoResourceRequest->add_ids(id);
+  mProtoResourceRequest->set_expected_total_size(
+      mProtoResourceRequest->expected_total_size() + size);
+  return true;
+}
+
+replay_service::ResourceRequest*
+ReplayConnection::ResourceRequest::release_to_proto() {
+  auto ptr = mProtoResourceRequest;
+  mProtoResourceRequest = nullptr;
+  return ptr;
+}
+
+ReplayConnection::ResourceRequest::ResourceRequest()
+    : mProtoResourceRequest(new replay_service::ResourceRequest()) {}
```

Comment:
Can we please put this in a unique_ptr. 
new/delete make me unhappy.

Reply:
done